consensus/consortium/v2/consortium.go

```diff
@@ -484,10 +484,10 @@ func (c *Consortium) snapshot(chain consensus.ChainHeaderReader, number uint64,
 				log.Trace("Loaded snapshot from disk", "number", number, "hash", hash.Hex())
 				break
 			}
+			header := chain.GetHeaderByHash(hash)
 
-			// get validators set from number
 			_, _, _, contract := c.readSignerAndContract()
-			validators, err := contract.GetBlockProducers(big.NewInt(0).SetUint64(number))
+			validators, err := contract.GetBlockProducers(header.ParentHash, new(big.Int).SetUint64(number))
```

Comment:
This does not use block hash because it does not reach the hardfork, so I think passing common.Hash{ } is enough